utf8proc.h

```diff
@@ -113,16 +113,24 @@ typedef bool utf8proc_bool;
 #endif
 #include <limits.h>
 
-#ifdef _WIN32
-#  ifdef UTF8PROC_EXPORTS
-#    define UTF8PROC_DLLEXPORT __declspec(dllexport)
+#ifndef UTF8PROC_BUILD_AS_LIB
+#  ifdef _WIN32
+#    ifdef UTF8PROC_EXPORTS
+#      define UTF8PROC_DLLEXPORT __declspec(dllexport)
+#    else
+#      define UTF8PROC_DLLEXPORT __declspec(dllimport)
+#    endif
+#  elif defined(__GNUC__) && __GNUC__ >= 4
+#    define UTF8PROC_DLLEXPORT __attribute__ ((visibility("default")))
 #  else
-#    define UTF8PROC_DLLEXPORT __declspec(dllimport)
+#    define UTF8PROC_DLLEXPORT
 #  endif
-#elif __GNUC__ >= 4
-#  define UTF8PROC_DLLEXPORT __attribute__ ((visibility("default")))
 #else
-#  define UTF8PROC_DLLEXPORT
+#  if defined(__GNUC__) && __GNUC__ >= 4
+#    define UTF8PROC_DLLEXPORT __attribute__ ((visibility("hidden")))
```

Comment:
I understand why you would want to hide symbols that you are not exporting.  We are already doing this by declaring internal functions as `static` (which works because all of our code is in a single `.c` file). But `UTF8PROC_DLLEXPORT` is used for *exported* functions.  Surely we don't want to hide *those* symbols?